conjure-lib/src/main/java/com/palantir/conjure/java/lib/DoubleArrayList.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.lib;
+
+import java.util.AbstractList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.RandomAccess;
+
+public final class DoubleArrayList extends AbstractList<Double> implements RandomAccess {
+    private int size = 0;
+    private double[] elements;
+
+    public DoubleArrayList() {
+        this(16);
+    }
+
+    private void resizeIfNecessary(int toAdd) {
+        if (size + toAdd > elements.length) {
+            elements = Arrays.copyOf(elements, Math.max(size + toAdd, elements.length * 2));
+        }
+    }
+
+    public DoubleArrayList(int initialCapacity) {
+        elements = new double[initialCapacity];
+    }
+
+    public DoubleArrayList(Collection<Double> collection) {
+        elements = new double[collection.size()];
+        Double[] array = collection.toArray(Double[]::new);
+        size = collection.size();
+        if (size != 0) {
+            for (int idx = 0; idx < size; idx++) {
+                elements[idx] = array[idx];
+            }
+        }
+    }
+
+    public DoubleArrayList(double[] doubleArray) {
+        elements = new double[doubleArray.length];
+        size = doubleArray.length;
+        if (size != 0) {
+            elements = Arrays.copyOf(doubleArray, size);
+        }
```

Comment:
This allocates twice. The ifrst allocation is thrown away. 
```suggestion
        size = doubleArray.length;
        elements = Arrays.copyOf(doubleArray, size);
```